aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1249,6 +1279,22 @@ private boolean s3KeyIdAccessKeyBothConfigured() {
     return (s3AccessKeyId == null) == (s3SecretAccessKey == null);
   }
 
+  private AwsCredentialsProvider credentialsProvider(String credentialsProviderClazz) {
+    try {
+      AwsCredentialsProvider provider =
+          DynMethods.builder("create").impl(credentialsProviderClazz).buildChecked().invoke(null);
```

Comment:
if it's always invoking null, then why not just initialize the client using an empty constructor? We might want to pass in configs of the client using properties with `client.credential-provider.` prefix.

Reply:
@jackye1995 I added support for both create() and create(Map<String, String) method from the provider class. If no properties set, it will call create(), else create(Map). Most of the AWS SDK Credential Providers are using create() method instead of constructor and thats why I followed this approach.